.github/workflows/ur-build-hw.yml

```diff
@@ -0,0 +1,132 @@
+name: UR - Build adapters on HW
+
+on:
+  workflow_call:
+    inputs:
+      adapter_name:
+        required: true
+        type: string
+      other_adapter_name:
+        required: false
+        type: string
+        default: ""
+      runner_name:
+        required: true
+        type: string
+      platform:
+        description: "Platform string, `UR_CTS_ADAPTER_PLATFORM` will be set to this."
+        required: false
+        type: string
+        default: ""
+      static_loader:
+        required: false
+        type: string
+        default: OFF
+      static_adapter:
+        required: false
+        type: string
+        default: OFF
+
+permissions:
+  contents: read
+
+env:
+  UR_LOG_CUDA: "level:error;flush:error"
+  UR_LOG_HIP: "level:error;flush:error"
+  UR_LOG_LEVEL_ZERO: "level:error;flush:error"
+  UR_LOG_NATIVE_CPU: "level:error;flush:error"
+  UR_LOG_OPENCL: "level:error;flush:error"
+
+jobs:
+  adapter_build_hw:
+    name: Build & CTS
+    # run only on upstream; forks won't have the HW
+    if: github.repository == 'intel/llvm'
+    strategy:
+      matrix:
+        adapter: [
+          {
+            name: "${{inputs.adapter_name}}",
+            other_name: "${{inputs.other_adapter_name}}",
+            platform: "${{inputs.platform}}",
+            static_Loader: "${{inputs.static_loader}}",
+            static_adapter: "${{inputs.static_loader}}"
+          }
+        ]
+        build_type: [Release]
+        compiler: [{c: gcc, cxx: g++}]
+
+    runs-on: ${{inputs.runner_name}}
+
+    steps:
+    # TODO: If UR is merged into llvm it will require changes:
+    # - checkout only llvm repo
+    # - configure UR project from local tree
+    #
+    # Also, the step of downloading DPC++ should be integrated somehow;
+    # either compile it or use cached binaries (if accessible)...?
```

Comment:
Nightly "release" maybe? We don't try re-use post-commit binaries (although artifacts might actually be available).

Reply:
Nightly would make sense, yeah. A related comment [here](https://github.com/intel/llvm/pull/16827/files#r1954752515).